### PR TITLE
Fix keepRuntimeAlive in the case where EXIT_RUNTIME=0 and noExitRuntime is not referenced

### PR DESCRIPTION
**Pull request by sbc100**
Essentially the `keepRuntimeAlive` was relying on the `noExitRuntime` variable being set based on `EXIT_RUNTIME` but when `noExitRuntime` was absent the `EXIT_RUNTIME` settings was being ignored and the runtime was exiting even though `EXIT_RUNTIME=0` was set (the default).

Fixes: #20636

### CODE REVIEW

**Review by sbc100 — COMMENTED** [review]
I updated the test to avoid the sleep.  PTAL

File: src/library_pthread.js [hunk 1]
```diff
@@ -188,7 +188,7 @@ var LibraryPThread = {
       // worker pool as an unused worker.
       worker.pthread_ptr = 0;
 
-#if ENVIRONMENT_MAY_BE_NODE && PROXY_TO_PTHREAD
```
Comment (kleisauke) [inline src/library_pthread.js:191]: Could you clarify why this change is necessary? AFAIK, all threads should already be weakly referenced by this point, this should only be required for the "main" thread in `PROXY_TO_PTHREAD` mode, see:
https://github.com/emscripten-core/emscripten/pull/19073#discussion_r1172276555

File: test/other/test_pthread_strict.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  pthread_detach(t);
+  emscripten_outf("thread created, returning from main");
+
+  // Make sure the thread output doesn't appear until after the above message
+  pthread_mutex_lock(&ready_lock);
+  ready = true;
+  pthread_cond_signal(&ready_cond);
+  pthread_mutex_unlock(&ready_lock);
+
+  // This needed under node to make sure the process doesn't exit before the
```
Comment (kripken) [inline test/other/test_pthread_strict.c:35]:
```suggestion
  // This is needed under node to make sure the process doesn't exit before the
```

File: test/other/test_pthread_strict.c [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  // thread is done.
+  _emscripten_thread_set_strongref(t);
```
Comment (kripken) [inline test/other/test_pthread_strict.c:37]: Do we expect users to need to do this?

Comment (sbc100) [reply test/other/test_pthread_strict.c:37]: No, this is just do that node doesn't exit while the background thread is running.  I could instead use a setTimeout loop instead I guess?

Comment (kripken) [reply test/other/test_pthread_strict.c:37]: I guess I was unaware that this was an issue. If the main thread exits and a background thread remains, then we need to keep the main thread alive in a special way, I guess, that is the underlying issue?

Comment (sbc100) [reply test/other/test_pthread_strict.c:37]: Right, its a little confusing.   Under normal circumstance, node will exit when the main function exits.  Running threads are not enough to keep node running.   The runtime has not exited in this case, the module is still available to be used, but node needs a reason to stay alive otherwise will exit the process.   

There is no equilient to this one the web, and this program will basically run forever on the web even without `_emscripten_thread_set_strongref`.. because there is no matching concept of "event loop is empty so exit" on the web.

I agree this is quite ugly here, so i might try to find a better way to keep node alive here.

Comment (kleisauke) [reply test/other/test_pthread_strict.c:37]: I think the canonical way is to remove the `pthread_detach()` call and replace `_emscripten_thread_set_strongref()` with `pthread_join()`, unless I'm missing something (perhaps this test case should not block the main thread?).

Comment (sbc100) [reply test/other/test_pthread_strict.c:37]: Hmm..there must be a reason I didn't do it that way... i'll need to go figure out why :)

Comment (sbc100) [reply test/other/test_pthread_strict.c]: Ah yes, the point of this test was to verify that when main returns it doesn't kill all the running threads.  If the thread was already joined there would be no point.

I re-wrote the test without threads at all PTAL

File: test/other/test_no_exit_runtime_strict.c [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+EM_JS(void, timeout_func, (), {
+  console.log("timeout_func: keepRuntimeAlive() ->", keepRuntimeAlive());
+  // once this timeout is done the node process should exit with 0
+});
+
+int main() {
+  int keep_alive = EM_ASM_INT({
+    setTimeout(timeout_func);
+    return keepRuntimeAlive();
+  }, emscripten_force_exit);
```
Comment (kripken) [inline test/other/test_no_exit_runtime_strict.c:18]: Why is a parameter passed in here?

File: test/other/test_no_exit_runtime_strict.c [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#include <assert.h>
+#include <emscripten/emscripten.h>
+#include <emscripten/em_js.h>
+#include <stdio.h>
+
+// Verify that `keepRuntimeAlive()` always returns true by default (i.e. when
+// EXIT_RUNTIME=0).
+
+EM_JS(void, timeout_func, (), {
+  console.log("timeout_func: keepRuntimeAlive() ->", keepRuntimeAlive());
```
Comment (kripken) [inline test/other/test_no_exit_runtime_strict.c:10]: Do you not need to ensure that keepRuntimeAlive is available in JS?

File: src/library_pthread.js (hunk 1 above)
Comment (sbc100) [reply src/library_pthread.js]: Reverted this part

File: test/other/test_no_exit_runtime_strict.c (hunk 4 above)
Comment (sbc100) [reply test/other/test_no_exit_runtime_strict.c:18]: Hmm.. I can't remember now.  Reverted.

**Review by kripken — APPROVED** [review]
